service/bfg/bfg.go

```diff
@@ -267,12 +259,16 @@ func (s *Server) handleRequest(parentCtx context.Context, bws *bfgWs, wsid strin
 	defer cancel()
 
 	select {
-	case <-s.requestLimiter:
+	case <-bws.requestLimiter:
+	case <-parentCtx.Done():
+		log.Debugf("handleRequest context done: %v", parentCtx.Err())
+		return
 	default:
-		log.Infof("Request limiter hit %v: %v", bws.addr, cmd)
-		<-s.requestLimiter
+		log.Infof("Request limiter hit %v (%v): %v", bws.addr, bws.sessionId, cmd)
```

Comment:
if the user has their requests limited, they're probably spamming us.  close their connection.

also, I believe closing this websocket will cause an error and thus will remove it from the sessions map